tests/validation/tests/v3_api/test_create_ha.py

```diff
@@ -121,17 +142,37 @@ def add_repo_create_namespace(repo=RANCHER_HELM_REPO):
     print(out)
 
 
-def install_rancher_self_signed(repo=RANCHER_HELM_REPO):
+def install_rancher(type=RANCHER_HA_CERT_OPTION, repo=RANCHER_HELM_REPO):
     helm_rancher_cmd = \
         export_cmd + " && helm_v3 install rancher " + \
         "rancher-" + repo + "/rancher " + \
         "--version " + RANCHER_CHART_VERSION + " " \
         "--namespace cattle-system " + \
         "--set hostname=" + RANCHER_HA_HOSTNAME
-
-    if RANCHER_IMAGE_TAG != "":
+        
+    if type == 'letsencrypt':
         helm_rancher_cmd = helm_rancher_cmd + \
-            " --set rancherImageTag=${RANCHER_IMAGE_TAG}"
+            " --set ingress.tls.source=letsEncrypt " + \
+            "--set letsEncrypt.email=" + RANCHER_LETSENCRYPT_EMAIL
+    elif type == 'byo-self-signed':
+        helm_rancher_cmd = helm_rancher_cmd + \
+            " --set ingress.tls.source=secret " + \
+            "--set privateCA=true"
+    elif type == 'byo-valid':
+        helm_rancher_cmd = helm_rancher_cmd + \
+            " --set ingress.tls.source=secret"
+    
+    if RANCHER_IMAGE_TAG != "" and RANCHER_IMAGE_TAG is not None:
+        helm_rancher_cmd = helm_rancher_cmd + \
+            " --set rancherImageTag=" + RANCHER_IMAGE_TAG
+
+    
+    if "byo-" in type:
+        valid_cert = True
+        if type == "byo-self-signed":
+            valid_cert = False
+
+        create_tls_secrets(valid_cert)
```

Comment:
Can you simplify the logic here since there are only 2 cases:
```
if type == "byo-self-signed":
	create_tls_secrets(valid_cert=False)
elif type == "byo-valid":
	create_tls_secrets(valid_cert=True)
```

Reply:
Done